### PR TITLE
Post/malicious usb creation and hunting

### PR DESCRIPTION
**Pull request by bittib010**
Need review

### CODE REVIEW

File: content/posts/2023-03-13-Hunt-for-and-craft-your-own-malicious-usb/index.mdx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+To protect yourself from malicious USB devices, it's important to follow a few simple rules. First and foremost, never plug in an untrusted device or one that you don't know where it came from. Always be wary of USB devices that you find in public places, as they may be designed to trick you into plugging them in.
+
+Secondly, make sure to sign out of your computer when you're done using it. This will help prevent unauthorized access by anyone who gains physical access to the device.
+
+# The basics of crafting a malicious USB drive
+
+There are several ways to get your hand on a malicious USB, the most popular is by far the Rubber Ducky by Hak5. The last few years there has been an increase in malicious cables, at least from my observations. The cable acts like a rubber ducky but looks like your normal USB charger cable. Another approach is to create one yourself using other popular microcontrollers. There are certain requirements for this to work but aiming for a Leonardo you are already there in terms of chipset that has this emulation feature.
```
Comment (bittib010) [inline content/posts/2023-03-13-Hunt-for-and-craft-your-own-malicious-usb/index.mdx:39]: What is a leonardo?

Change the title? or deepen the insight into the crafting process?

File: content/posts/2023-03-13-Hunt-for-and-craft-your-own-malicious-usb/index.mdx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+Way back when I started researching the topics of malicious USB's i made one myself that was able to be used regardless of the OS I wanted to attack. To achieve this, it had a few buttons, 3x2. Where the first row of buttons was for OS (Windows, MacOS and Linux), the next row was for filetypes to extract (text/documents, videos and pictures). Pressing a combination of OS button and filetype, files were then sent to a USB storage located in the same hub as the Arduino.
```
Comment (bittib010) [inline content/posts/2023-03-13-Hunt-for-and-craft-your-own-malicious-usb/index.mdx:41]: to send the files, a command had to be run. Point this out.

Comment (bittib010) [reply content/posts/2023-03-13-Hunt-for-and-craft-your-own-malicious-usb/index.mdx:41]: Also, make a point of the pot in use

File: content/posts/2023-03-13-Hunt-for-and-craft-your-own-malicious-usb/index.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+- UsbDriveUnmounted
+- UsbDriveLetterChanged
+
+But making it easier, does not equal to easy. Attackers always look for ways to hide their steps, and by changing and spoofing the USB PID and VID (Product ID and Vendor ID) they make it harder to be detected. But this is by far the most effective way also, as this means that there is an additional step in creating such a malicious USB. I finally resorted into this approach as I tried quite a few different approaches into detecting the few scripts I was running on my device. First of all, I tried to look at the ```UsbDriveMounted``` and set a timegap between that and ```UsbDriveUnmounted``` to look for all process creations and PowerShell commands created during that connection time. But as you may think, that was ineffiecient. It yielded a result not worth my time to dig through. I tried a few variations of this before I saw that I need to filter on known or potentially malicious VIDs and PIDs. And then I found a great resource to go look for such devices based on what kinds of microchip they have. Not all microchips have the ability to emulate a keyboard - those who do, may be used in these types of attacks. Here is a draft:
```
Comment (bittib010) [inline content/posts/2023-03-13-Hunt-for-and-craft-your-own-malicious-usb/index.mdx:58]: "But this is by far the most effective way also,"

"this" - replace "with hunting for these"

Add a potential answer to limit the endless amount of results from the tested query where you filter on known powershell modules. Maybe a link to such as well (one by Alex Verboon i think)

**Review by FRGulli — APPROVED** [review]
Markdown formatting looks correct!

**Pull request merged**